packages/cactus-plugin-keychain-vault/src/test/typescript/integration/plugin-keychain-vault.test.ts

```diff
@@ -35,79 +61,182 @@
+
+  const key1 = uuidv4();
+  const value1 = uuidv4();
+
+  const hasPrior1 = await plugin.has(key1);
+
+  t.false(hasPrior1, "hasPrior === false OK");
+
+  await plugin.set(key1, value1);
+
+  const hasAfter1 = await plugin.has(key1);
+  t.true(hasAfter1, "hasAfter === true OK");
+
+  const valueAfter1 = await plugin.get(key1);
+  t.ok(valueAfter1, "valueAfter truthy OK");
+  t.equal(valueAfter1, value1, "valueAfter === value OK");
+
+  await plugin.delete(key1);
+
+  const hasAfterDelete1 = await plugin.has(key1);
+  t.false(hasAfterDelete1, "hasAfterDelete === false OK");
+
+  const valueAfterDelete1 = await plugin.get(key1);
+  t.notok(valueAfterDelete1, "valueAfterDelete falsy OK");
+
+  {
+    const res = await apiClient.getPrometheusExporterMetricsV1();
+    const promMetricsOutput =
+      "# HELP " +
+      K_CACTUS_KEYCHAIN_VAULT_TOTAL_KEY_COUNT +
+      " Total keys present in vault\n" +
+      "# TYPE " +
+      K_CACTUS_KEYCHAIN_VAULT_TOTAL_KEY_COUNT +
+      " gauge\n" +
+      K_CACTUS_KEYCHAIN_VAULT_TOTAL_KEY_COUNT +
+      '{type="' +
+      K_CACTUS_KEYCHAIN_VAULT_TOTAL_KEY_COUNT +
+      '"} 0';
+    t.ok(res);
+    t.ok(res.data);
+    t.equal(res.status, 200);
+    t.true(
+      res.data.includes(promMetricsOutput),
+      "Total Key Count 0 recorded as expected. RESULT OK",
+    );
+  }
+
+  const key2 = uuidv4();
+  const value2 = uuidv4();
+
+  const hasPrior2 = await plugin.has(key2);
+
+  t.false(hasPrior2, "hasPrior === false OK");
+
+  await plugin.set(key2, value2);
+
+  const hasAfter2 = await plugin.has(key2);
+  t.true(hasAfter2, "hasAfter === true OK");
+
+  const valueAfter2 = await plugin.get(key2);
+  t.ok(valueAfter2, "valueAfter truthy OK");
+  t.equal(valueAfter2, value2, "valueAfter === value OK");
+
+  {
+    const res = await apiClient.getPrometheusExporterMetricsV1();
+    const promMetricsOutput =
+      "# HELP " +
+      K_CACTUS_KEYCHAIN_VAULT_TOTAL_KEY_COUNT +
+      " Total keys present in vault\n" +
+      "# TYPE " +
+      K_CACTUS_KEYCHAIN_VAULT_TOTAL_KEY_COUNT +
+      " gauge\n" +
+      K_CACTUS_KEYCHAIN_VAULT_TOTAL_KEY_COUNT +
+      '{type="' +
+      K_CACTUS_KEYCHAIN_VAULT_TOTAL_KEY_COUNT +
+      '"} 1';
+    t.ok(res);
+    t.ok(res.data);
+    t.equal(res.status, 200);
+    t.true(
+      res.data.includes(promMetricsOutput),
+      "Total Key Count 1 recorded as expected. RESULT OK",
+    );
+  }
+
+  t.end();
+});
+
+test.skip("rotateEncryptionKeys() fails fast", async (t: Test) => {
+  const options: IPluginKeychainVaultOptions = {
+    instanceId: uuidv4(),
+    keychainId: uuidv4(),
+    endpoint: "http://127.0.0.1:9200",
+    token: "root",
+  };
+  const plugin = new PluginKeychainVault(options);
+
+  const promise = plugin.rotateEncryptionKeys();
+  const expected = /not implemented/;
+  await t.rejects(promise, expected, "rotateEncryptionKeys() rejects OK");
+
+  t.end();
+});
+
+test.skip("getEncryptionAlgorithm() returns null", (t: Test) => {
```

Comment:
@jagpreetsinghsasan Same idea for skipping tests as for the commented code: please leave a TODO or FIXME comment and create a GH issue for those as well so that in the future when we have more time someone can come around and fix these. And once again sorry for not setting the right example here, I've been setting tests to skip myself as well without explanation but it should always be explained right there in the code in a comment why the test must be skipped.